Mage.Sets/src/mage/cards/r/Recantation.java

```diff
@@ -38,7 +40,7 @@ public Recantation(UUID ownerId, CardSetInfo setInfo) {
         Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, effect, new ManaCostsImpl<>("{U}"));
         ability.addCost(new SacrificeSourceCost());
         ability.addTarget(new TargetPermanent(0, 0, new FilterPermanent("up to X target permanents, where X is the number of verse counters on {this}."), false));
```

Comment:
max 0 instead of 1, does it matter? (applies to all adjusters for verse counters)

Reply:
It does not. The only one that matters is the min value (0 for "up to", anything else for exactly that many)